src/extra_files.c

```diff
@@ -90,29 +90,36 @@ static int qsort_helper(const void *A, const void *B)
 	return strcmp(((struct filerecord *)A)->filename, ((struct filerecord *)B)->filename);
 }
 
-static void handle(const char *filename, bool is_dir, bool fix)
+static bool handle(const char *filename, bool is_dir, bool fix)
 {
 	char *temp;
+	bool ret = true;
 
 	if (fix) {
 		string_or_die(&temp, "%s%s", path_prefix, filename);
 		fprintf(stderr,
 			is_dir ? "REMOVING DIR %s/\n" : "REMOVING %s\n",
 			filename);
 		if (remove(temp)) {
-			fprintf(stderr, "Removing %s failed: %s", temp, strerror(errno));
+			fprintf(stderr, "Removing %s failed: %s\n", temp, strerror(errno));
+			ret = false;
 		}
 		free_string(&temp);
 	} else {
 		printf("%s%s\n", filename, is_dir ? "/" : "");
+		ret = false;
 	}
+
+	/* return true if the function deleted a file or directory, false otherwise */
```

Comment:
I would put this comment just before the 'static bool handle' definition line, and then question if it was correct. In the case that you are attempting to remove a directory but fail you may well delete some files but still return false.

Perhaps a better comment would be "return true if the function deletes the specified filename, otherwise false".

Failure examples are hard, because you are running as root, but could happen. A race condition where a new file is added to a directory whilst `remove` is busy removing a subtree might cause the eventual removal of the directory to fail as it is not empty. A read-only file system might be mounted on a sub directory.

Thank you for the struct file_counts change - I think it is good.

Reply:
I really appreciate your comments @icarus-sparry I/m learning so much. Thanks for reviewing. I'll update the comment as you suggested.